htdocs/salaries/card.php

```diff
@@ -146,14 +146,12 @@
 		if ($ret > 0)
 		{
 			$db->commit();
-			header("Location: list.php");
-			exit;
+			unset($_POST['fk_user']);
 		}
 		else
 		{
 			$db->rollback();
 			setEventMessages($object->error, $object->errors, 'errors');
-			$action = "create";
```

Comment:
Why removing this line ?
It is used so if there is an error when creating the salary payment (for example because a mandatory field is not filled), we stay on the form to create, so user has a chance to fix what is missing and can easily resubmit the form).

Reply:
It is duplicate of line 158. $action is always set to 'create', no need to set it separately for failure case. Just a cleanup of redundant code, it functions exactly as before.